BSP/buttons/buttons.c

```diff
@@ -0,0 +1,41 @@
+#include "buttons.h"
+#include <stm32l4xx_hal.h>
+
+/**
+ * @brief list of BUTTON ports in same order as @ref button_e
+ */
+static GPIO_TypeDef *const buttonPortTable[NB_BUTTONS] = {
```

Comment:
Good idea. I will do so for every PR. 
Because with INTERFACE lib I think it's impossible to have private header files I will put the typedef into each .c file